bokeh/charts/builder.py

```diff
@@ -547,6 +573,47 @@ def generate_help(cls):
 
         return help_str
 
+    @staticmethod
+    def _sort_legend(legend_sort_field, legend_sort_direction, legends, attributes):
+        """Sort legends sorted by looping though sort_legend items (
+        see :attr:`Builder.sort_legend` for more details)
+        """
+        if legend_sort_field:
+            if len(attributes[legend_sort_field].columns) > 0:
+
+                # TODO(fpliger): attributes should be consistent and not
+                #               need any type checking but for
+                #               the moment it is not, specially when going
+                #               though a process like binning or when data
+                #               is built for HeatMap, Scatter, etc...
+                item_order = [x[0] if isinstance(x, tuple) else x
+                    for x in attributes[legend_sort_field].items]
+
+                item_order = [str(x) if not isinstance(x, string_types)
+                    else x for x in item_order]
+
+                def foo(leg):
+                    return item_order.index(leg[0])
+                reverse = legend_sort_direction == 'descending'
+                return list(sorted(legends, key=foo, reverse=reverse))
+
+        return legends
+
+    @property
+    def sort_legend(self):
+        warnings.warn("Chart property 'sort_legend' was deprecated in 0.12 \
+            and will be removed in the future.")
+        return [(self.legend_sort_field, self.legend_sort_direction)]
+
+    @sort_legend.setter
+    def sort_legend(self, value):
+        warnings.warn("Chart property 'sort_legend' was deprecated in 0.12 \
+            and will be removed in the future.")
```

Comment:
need to point users to the replacement. Also not certain, but worth seeing if https://github.com/bokeh/bokeh/blob/master/bokeh/util/deprecate.py works with properties.